### PR TITLE
feat: python workflow sample code

### PR DESCRIPTION
**Pull request by Eileen-Yu**
resolve https://github.com/diagridio/issues/issues/4384
To test it locally:
1. fetch latest CLI repo code, and build it by `make dev`
2. create a dir called `workflow-python-project` wherever you like, and copy & paste all files in `workflow/python` in this PR into that dir you created (this is to skip the download step)
3. run `diagrid project quickstart --type workflow --language python (--container)` to test it

### CODE REVIEW

**Review by Eileen-Yu — COMMENTED** [review]
ready, need merge this first then update commit sha in CLI

File: workflow/python/requirements.txt [hunk 1]
```diff
@@ -0,0 +1,9 @@
+dapr==1.13.0
```
Comment (sicoyle) [inline workflow/python/requirements.txt:1]: can we use the 1.14 rc somehow here?

Comment (Eileen-Yu) [reply workflow/python/requirements.txt]: updated!

File: workflow/python/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+# Create a Python virtual environment
+RUN python -m venv venv
+
+# List contents to verify creation
+RUN ls -la venv/bin/
+
+# Install Diagrid CLI 
+RUN curl -o- https://downloads.diagrid.io/cli/install.sh | bash
+RUN mv ./diagrid /usr/local/bin/diagrid || echo "Diagrid binary not found!"
```
Comment (detzgk) [inline workflow/python/Dockerfile:18]: Having this "|| echo" here will make the RUN statement succeed no matter what, and docker will keep going with the build, so I'm not sure it's a good idea.

File: workflow/python/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+RUN chmod +x /usr/local/bin/diagrid
+
+# Copy sample code
+COPY . .
+
+# Activate virtual environment and update pip and install certifi
+RUN . venv/bin/activate && pip install --upgrade pip && pip install certifi
+# Install Python dependencies
+RUN . venv/bin/activate && pip install --no-cache-dir -r requirements.txt
```
Comment (detzgk) [inline workflow/python/Dockerfile:27]: For single commands you can just run the venv version of pip directly:

RUN ./venv/bin/pip install --upgrade pip
RUN ./venv/bin/pip install certifi
RUN ./venv/bin/pip install --no-cache-dir -r requirements.txt

File: workflow/python/requirements.txt [hunk 4]
```diff
@@ -0,0 +1,9 @@
+dapr==1.14.0-rc.4
+dapr-ext-workflow
+fastapi==0.109.1
+grpcio==1.62.0
```
Comment (detzgk) [inline workflow/python/requirements.txt:4]:
```suggestion
grpcio==1.63.0
```
Avoids another warning in the logs: 
```== APP - workflow-app == /Users/eli/Source/workflow-python-project/diagrid-venv/lib/python3.12/site-packages/dapr/proto/runtime/v1/dapr_pb2_grpc.py:24: RuntimeWarning: The grpc package installed is at version 1.62.0, but the generated code in dapr/proto/runtime/v1/dapr_pb2_grpc.py depends on grpcio>=1.63.0. Please upgrade your grpc module to grpcio>=1.63.0 or downgrade your generated code using grpcio-tools<=1.62.0. This warning will become an error in 1.65.0, scheduled for release on June 25, 2024.```

**Comment by detzgk** [comment]
@Eileen-Yu I tested with `dapr==1.14.0rc1` because it's the one I see available on pypi and everything seemed to work OK. The instructions at the end of the project creation seem a little fuzzy to me. We tell people to create their own venv with `python -m venv venv` but we've already created one in `diagrid-venv`, is that on purpose? We then tell them to pip install but without explicitly telling them to activate the venv. I don't know whether we're targeting expert Pythonistas who are going to know how to work with venvs, or a broader audience that may not be familiar with them.

File: workflow/python/Dockerfile (hunk 2 above)
Comment (Eileen-Yu) [reply workflow/python/Dockerfile:18]: thx for catching!

File: workflow/python/Dockerfile (hunk 3 above)
Comment (Eileen-Yu) [reply workflow/python/Dockerfile]: thx for letting me know!

**Comment by Eileen-Yu** [comment]
> @Eileen-Yu I tested with `dapr==1.14.0rc1` because it's the one I see available on pypi and everything seemed to work OK. The instructions at the end of the project creation seem a little fuzzy to me. We tell people to create their own venv with `python -m venv venv` but we've already created one in `diagrid-venv`, is that on purpose?

Thx for verifying that @detzgk ! I updated the dapr version.

Yeah the instruction is only for `host version`, when user run `diagrid project quickstart --type workflow --language python` without the `container` flag, then it means they choose to set up the dev env by themselves, so they need to follow the setup instructions.

The  `diagrid-venv`  is for `--container` flag usage, in that case we set up the container and handle all the env setup for the users. In such case they will only got the log without the instruction:
```
❯ dg project quickstart --type workflow --language python --container
○  Checking application dependency for python...
python language dependency is installed.
○  Checking docker status to run the quickstart in container...
✓  Docker is installed and running.
○  Downloading sample application code from https://github.com/diagridio/catalyst-quickstarts...
○  workflow-python-project already exists, skipping download.
○  Building docker image...
✓  Docker image is built successfully.
Output:
✓  To run the quickstart in container, run `docker run -v ~/.diagrid/creds:/root/.diagrid/creds -it -p 5001:5001 workflow-python-container`
```

File: workflow/python/Dockerfile (hunk 3 above)
Comment (Eileen-Yu) [reply workflow/python/Dockerfile]: actually just found we may remove these lines as the virtual env will be set up in entrypoint.sh

**Comment by Eileen-Yu** [comment]
ready 🚀

File: workflow/python/requirements.txt [hunk 5]
```diff
@@ -0,0 +1,9 @@
+dapr==1.14.0rc1
```
Comment (sicoyle) [inline workflow/python/requirements.txt:1]: I think there is now an rc2 that might be nice to grab :)

**Review by detzgk — APPROVED** [review]
LGTM, everything seems to work out of the box. We do end up creating a diagrid-venv no matter what execution mode the user chooses, but I think it's just to run the scaffold.py

**Review by sicoyle — APPROVED** [review]
🎉

File: workflow/python/requirements.txt (hunk 5 above)
Comment (Eileen-Yu) [reply workflow/python/requirements.txt:1]: it seems for PyPI the rc1 is the latest 🤔
```
❯ pip install -r requirements.txt
ERROR: Ignored the following yanked versions: 1.8.0
ERROR: Could not find a version that satisfies the requirement dapr==1.14.0rc2 (from versions: 0.5.0a0, 0.6.0a0, 0.6.0a1, 0.7.0a0, 0.8.0a0, 1.0.0rc2, 1.0.0rc3, 1.0.0rc4, 1.0.0, 1.1.0rc1, 1.1.0, 1.2.0rc1, 1.2.0, 1.3.0rc1, 1.3.0, 1.4.0rc1, 1.4.0, 1.4.1, 1.5.0rc1, 1.5.0, 1.6.0rc1, 1.6.0, 1.6.1, 1.7.0, 1.8.0rc1, 1.8.0rc2, 1.8.1, 1.8.2rc1, 1.8.2, 1.8.3rc1, 1.8.3, 1.9.0rc2, 1.9.0rc4, 1.9.0, 1.9.1, 1.10.0rc1, 1.10.0, 1.11.0, 1.12.0, 1.12.1, 1.13.0rc1, 1.13.0, 1.14.0rc1)
ERROR: No matching distribution found for dapr==1.14.0rc2
```

File: workflow/python/run.py [hunk 6]
```diff
@@ -0,0 +1,138 @@
...
+                break
+        
+        print(f"Attempt {attempt}: Current status of {appid_name}: {status}")
+        if status and (status.lower() == "ready" or status.lower() == "available"):
+            break
+        if attempt == max_attempts:
+            print(f"Max attempts reached. {appid_name} is not ready.")
+            sys.exit(1)
+        
+        print("Waiting for 10 seconds...")
```
[the remark this answers is not in the document]
Comment (Eileen-Yu) [reply workflow/python/run.py]: sure, updated!

**Comment by Eileen-Yu** [comment]
im gonna merge this, and certainly can iter it for any further improvement in followup PRs

File: workflow/python/entrypoint.sh [hunk 7]
```diff
@@ -0,0 +1,14 @@
+#!/bin/bash
+# Create a project and set up the environment
+python3 run.py --project-name workflow-python-project-container --config-file "$CONFIG_FILE" --is-container
+
+# Install deps
+python3 -m venv venv
```
Comment (Eileen-Yu) [inline workflow/python/entrypoint.sh:6]: the python virtual env is project-scoped, should seldom would hit the 'already exist' issue (though we already have logic to check that). so here just all use the standard naming 'venv'

**Pull request merged**